test/table/test_table.py

```diff
@@ -480,3 +480,18 @@ def test_table_with_no_headers_nor_horizontal_lines(tmp_path):  # discussion 924
     assert_pdf_equal(
         pdf, HERE / "table_with_no_headers_nor_horizontal_lines.pdf", tmp_path
     )
+
+
+def test_table_with_multiple_headings_and_pagebreak(tmp_path):
```

Comment:
Could you please also add unit tests for the cases you handled when both `first_row_as_headings` & `num_heading_rows` are provided?